### ISSUE TITLE
[5.0] Per-route VerifyCsrfToken middleware fails: "Call to a member function setCookie() on null"

### ISSUE THREAD

**Issue by berniegp** [laravel/framework#11942]
Using a vanilla v5.0.34 install:

1- do the following changes to app/Http/Kernel.php to remove the global CSRF middleware
```
protected $middleware = [
	'Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode',
	'Illuminate\Cookie\Middleware\EncryptCookies',
	'Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse',
	'Illuminate\Session\Middleware\StartSession',
	'Illuminate\View\Middleware\ShareErrorsFromSession',
	// 'App\Http\Middleware\VerifyCsrfToken',
];

protected $routeMiddleware = [
	'auth' => 'App\Http\Middleware\Authenticate',
	'auth.basic' => 'Illuminate\Auth\Middleware\AuthenticateWithBasicAuth',
	'guest' => 'App\Http\Middleware\RedirectIfAuthenticated',
	'csrf' => 'App\Http\Middleware\VerifyCsrfToken',
];
```

2- in App\Http\Controllers\WelcomeController.php, add the CSRF middleware:
```
public function __construct()
{
	$this->middleware('guest');
	$this->middleware('csrf');
}
```

3- run phpunit tests or visit the home page and get the following:

```
FatalErrorException in VerifyCsrfToken.php line 76:
Call to a member function setCookie() on null
```

**Comment by berniegp** [comment]
Cannot reproduce in 5.1.

I suspect this difference in ControllerDispatcher to be related:

5.0:
```
		return (new Pipeline($this->container))
	                ->send($request)
	                ->through($middleware)
	                ->then(function($request) use ($instance, $route, $method)
					{
						return $this->call($instance, $route, $method);
					});
```

5.1:
```
        return (new Pipeline($this->container))
                    ->send($request)
                    ->through($middleware)
                    ->then(function ($request) use ($instance, $route, $method) {
                        return $this->router->prepareResponse(
                            $request, $this->call($instance, $route, $method)
                        );
                    });
```

Note the added `prepareResponse` method call.

**Comment by arrilot** [comment]
5.0 is closed for changes.

**Comment by berniegp** [comment]
Please allow me to quote a passage from the migration docs:

>Upgrading To 5.0 From 4.2
>To start, install a new Laravel 5.0 application into a fresh directory in your local environment. Do not install any versions newer than 5.0 yet, since we need to complete the migration steps for 5.0 first. We'll discuss each piece of the migration process in further detail below.

This issue completely blocked my upgrade because I followed the advice in the upgrade guide. If no bug fixes are coming in 5.0, I suggest to at least remove this advice from the upgrade doc.
